packages/playback-core/src/text-tracks.ts

```diff
@@ -42,17 +41,29 @@ export function setupTextTracks(
       return textTrack.id && textTrack.mode === 'showing' && ['subtitles', 'captions'].includes(textTrack.kind);
     });
 
+    if (!showingTrack) return;
+
+    const currentHlsTrack = hls.subtitleTracks[hls.subtitleTrack];
+
     // If hls.subtitleTrack is -1 or its id changed compared to the one that is showing load the new subtitle track.
-    const hlsTrackId = `${hls.subtitleTracks[hls.subtitleTrack]?.type.toLowerCase()}${hls.subtitleTrack}`;
-    if (showingTrack && (hls.subtitleTrack < 0 || showingTrack?.id !== hlsTrackId)) {
-      const idx = hls.subtitleTracks.findIndex(({ lang, name, type }) => {
-        return lang == showingTrack.language && name === showingTrack.label && type.toLowerCase() === showingTrack.kind;
+    const hlsTrackId = !currentHlsTrack
+      ? undefined
+      : currentHlsTrack.default
+        ? 'default'
+        : `${hls.subtitleTracks[hls.subtitleTrack].type.toLowerCase()}${hls.subtitleTrack}`;
+
+    if (hls.subtitleTrack < 0 || showingTrack?.id !== hlsTrackId) {
+      const idx = hls.subtitleTracks.findIndex(({ lang, name, type, default: defaultTrack }) => {
+        return (
+          (showingTrack.id === 'default' && defaultTrack) ||
+          (lang == showingTrack.language && name === showingTrack.label && type.toLowerCase() === showingTrack.kind)
+        );
       });
       // After the subtitleTrack is set here, hls.js will load the playlist and CUES_PARSED events will be fired below.
       hls.subtitleTrack = idx;
     }
 
-    if (showingTrack && showingTrack?.id === hlsTrackId) {
+    if (showingTrack?.id === hlsTrackId) {
```

Comment:
what if undefined === undefined ?
that showingTrack null check was needed I think

Reply:
we early bail now